pkg/serverless/trace/propagation/carriers.go

```diff
@@ -201,6 +213,47 @@ func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
 	return carrier, nil
 }
 
+// handleEventBridgeThroughSNS builds payload carrier for an EventBridge
+// event wrapped in an SNS message.
+func handleEventBridgeThroughSNS(eventBridgeEvent map[string]interface{}) (tracer.TextMapReader, error) {
+	detail, ok := eventBridgeEvent["detail"].(map[string]interface{})
+	if !ok {
+		return nil, errorUnsupportedPayloadType
+	}
+
+	datadogInfo, ok := detail[datadogTraceHeader].(map[string]interface{})
+	if !ok {
+		return nil, errorNoDDContextFound
+	}
+
+	carrier := make(tracer.TextMapCarrier)
+	for key, value := range datadogInfo {
+		if strValue, ok := value.(string); ok {
+			carrier[key] = strValue
+		}
+	}
+	return carrier, nil
```

Comment:
Is there a better way to do this

Reply:
@purple4reina might have a good answer for this

Reply:
Yes. Here's another option: https://github.com/DataDog/datadog-agent/compare/main...rey.abolofia%2Fnhulston

Notice one thing about this code. I rely on go typing and json unmarshalling to do a lot of the work for me. I change the `Detail` field on the `EventBridgeEvent` to a `map[string]map[string]string`. Since that's the type you're wanting, just force it into that from the start. It's also just fun to say, so a little bonus benefit there too 😉